src/main/target/NOX/target.h

```diff
@@ -0,0 +1,119 @@
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this software.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#define USE_TARGET_CONFIG
+
+#define TARGET_BOARD_IDENTIFIER "NOX1"
+#define USBD_PRODUCT_STRING "NoxF4V1"
+
+#define LED0_PIN                PA4
+
+#define BEEPER                  PC13
+#define BEEPER_INVERTED
+
+#define INVERTER_PIN_UART2      PC14
+
+#define USE_ACC
+#define USE_ACC_SPI_MPU6500
+#define USE_ACC_SPI_MPU6000
+
+#define USE_GYRO
+#define USE_GYRO_SPI_MPU6500
+#define USE_GYRO_SPI_MPU6000
+
+#define MPU6500_CS_PIN          PB12
+#define MPU6500_SPI_INSTANCE    SPI2
+
+#define MPU6000_CS_PIN          PB12
+#define MPU6000_SPI_INSTANCE    SPI2
+
+#define USE_EXTI
+#define MPU_INT_EXTI            PA8
+#define USE_MPU_DATA_READY_SIGNAL
+
+#define USE_BARO
+#define USE_BARO_BMP280
+#define USE_BARO_SPI_BMP280
+#define BMP280_SPI_INSTANCE     SPI2
+#define BMP280_CS_PIN           PA9
+
+#define USE_OSD
+#define USE_MAX7456
+#define MAX7456_SPI_INSTANCE    SPI2
+#define MAX7456_SPI_CS_PIN      PA10
+#define MAX7456_SPI_CLK         (SPI_CLOCK_STANDARD*2)
+#define MAX7456_RESTORE_CLK     (SPI_CLOCK_FAST)
+
+#define USE_VCP
+//#define VBUS_SENSING_PIN PC5
+
+#define USE_UART1
+#define UART1_RX_PIN            PB7
+#define UART1_TX_PIN            PB6
+
+#define USE_UART2
+#define UART2_RX_PIN            PA3
+#define UART2_TX_PIN            PA2
+
+#define USE_SOFTSERIAL1
+#define SOFTSERIAL1_RX_PIN      PA2    // Backdoor timer on UART2_TX, used for ESC telemetry
+#define SOFTSERIAL1_TX_PIN      PA2    // Workaround for softserial not initializing with only RX
+
+#define USE_SOFTSERIAL2
+#define SOFTSERIAL2_RX_PIN      NONE
+#define SOFTSERIAL2_TX_PIN      NONE
+
+#define SERIAL_PORT_COUNT       5 //VCP, USART1, USART2, SOFTSERIAL1, SOFTSERIAL2
+
+#define USE_ESCSERIAL
+#define ESCSERIAL_TIMER_TX_HARDWARE 0 // PPM
+
+#define USE_SPI
+#define USE_SPI_DEVICE_1
+#define SPI1_SCK_PIN            PB3
+#define SPI1_MISO_PIN           PB4
+#define SPI1_MOSI_PIN           PB5
+
+#define USE_SPI_DEVICE_2
+#define SPI2_SCK_PIN            PB13
+#define SPI2_MISO_PIN           PB14
+#define SPI2_MOSI_PIN           PB15
+
+#define USE_ADC
+#define CURRENT_METER_ADC_PIN   NONE // PA6 Available from TP33
+#define VBAT_ADC_PIN            PA5  // 11:1 (10K + 1K) divider
+
+#define DEFAULT_VOLTAGE_METER_SOURCE VOLTAGE_METER_ESC
+#define DEFAULT_CURRENT_METER_SOURCE CURRENT_METER_ESC
+
+#define SERIALRX_UART           SERIAL_PORT_USART2
+#define ESC_SENSOR_UART         SERIAL_PORT_SOFTSERIAL1
```

Comment:
Can we please **not** do this. While it is cute, it is also cluttering up target definitions, making them harder to maintain , because it introduces a third category 'needed only for this target' to the list of 'needed' and 'not needed'. Just stick your settings into `config.c`.

Reply:
Yeah, I thought about it, but did it because it was `cute` 😉

Reply:
It breaks one of Martin's rules of programming: no cute code. (Also no cute names.)

Reply:
BABEFACE?

Reply:
OK, you got me there. :smiley:

Reply:
This did not even trigger my 'cute is tempting' smell detector, but it brought up memories of wading through lists of unneeded and unneeded-looking defines in `target.h` files, trying to separate the two.